### PR TITLE
feat: component check list

### PR DESCRIPTION
**Pull request by bashirkarimi**

### CODE REVIEW

File: packages/styles/src/components/list.scss
[the remark this answers is not in the document]
Comment (bashirkarimi) [reply packages/styles/src/components/list.scss:1]: The `bottom` and `left` which have `0` value are reseting the `reboot` and can not be changed. The rest are changed accordingly.

**Comment by oliverschuerch** [comment]
@bashirkarimi create a :after element, place it the same way as the :before element.
Set the background-color of it through the icon and define the icon as a mask-image (there is a mixin for that).

File: packages/styles/src/components/list.scss [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    position: relative;
+    padding-inline-start: calc(
+      #{tokens.get('post-list-checks', 'spacing-gap-inline-text'),
+      } + #{$sizing-icon}
+    );
+
+    &::before {
+      content: '';
+      position: absolute;
+      left: 0;
```
Comment (gfellerph) [inline packages/styles/src/components/list.scss:27]: The check is not centrally aligned with the first line of text of the list item
image

File: packages/styles/src/components/list.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
+@use './../themes/bootstrap/core' as *;
```
Comment (gfellerph) [inline packages/styles/src/components/list.scss:1]: We're going to remove bootstrap, this import can't stay here. Please replace every reference to bootstrap with internal functionality.

File: packages/styles/src/components/list.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      background-repeat: no-repeat;
+      background-position-x: 50%;
+      background-position-y: center;
+      background-color: tokens.get('post-list-checks', 'color-icon-bg');
+      border-radius: 50%;
+      background-size: calc(#{$sizing-icon} - #{$padding-icon} * 2);
+      background-image: url('#{icons.get-colored-svg-url("2105", "white")}'); // 'white' should be changed with variable
+    }
+
+    @include media-breakpoint-down(sm) {
```
Comment (gfellerph) [inline packages/styles/src/components/list.scss:40]: Please replace this bootstrap mixin with it's counterpart from https://github.com/swisspost/design-system/blob/main/packages/styles/src/mixins/_media.scss

File: packages/styles/src/components/list.scss (hunk 2 above)
Comment (bashirkarimi) [reply packages/styles/src/components/list.scss]: It is removed.

File: packages/styles/src/components/list.scss (hunk 3 above)
Comment (bashirkarimi) [reply packages/styles/src/components/list.scss]: It is replaced.

File: packages/styles/src/components/list.scss (hunk 1 above)
Comment (bashirkarimi) [reply packages/styles/src/components/list.scss]: The size of icon is fixed.

**Review by gfellerph — CHANGES_REQUESTED** [review]
On mobile, the checklist is now misaligned.

image

And in high contrast mode, the round checks disappear:
image

Can you add a rule for high contrast mode like

```CSS
@include utilities.high-contrast-mode() {
    background-color: CanvasText;
    color: Canvas;
}
```

**Comment by bashirkarimi** [comment]
> On mobile, the checklist is now misaligned.
> 
> image
> And in high contrast mode, the round checks disappear: image
> 
> Can you add a rule for high contrast mode like
> 
> ```css
> @include utilities.high-contrast-mode() {
>     background-color: CanvasText;
>     color: Canvas;
> }
> ```



> On mobile, the checklist is now misaligned.
> 
> image
> And in high contrast mode, the round checks disappear: image
> 
> Can you add a rule for high contrast mode like
> 
> ```css
> @include utilities.high-contrast-mode() {
>     background-color: CanvasText;
>     color: Canvas;
> }
> ```

**Comment by bashirkarimi** [comment]
> On mobile, the checklist is now misaligned.
> 
> image
> And in high contrast mode, the round checks disappear: image
> 
> Can you add a rule for high contrast mode like
> 
> ```css
> @include utilities.high-contrast-mode() {
>     background-color: CanvasText;
>     color: Canvas;
> }
> ```

The high contrast mode is added.
The Tokens for icon size does not change on breakpoints, currently. Therefore, the size of icon is the same for all screen sizes and it overflow to bottom on mobile.
I asked Alessio about tokens changing on breakpoints and here is his reply: 

> The font-size is atm device dependent, meaning it pulls different numbers from the semantical level based on what screensice people are on. Not sure if it will just stay that way or if they want to switch it up once we got figma enterprise as we then can have 40 different breakpoints compared to 4 right now. Therefore as it is now: Desktop ≥ 1024, Tablet 600-1023, Mobile <600

**Comment by gfellerph** [comment]
There should be a way to always align the checkmark on the first line of content, no matter the size of the checkmark or the text.

**Review by gfellerph — CHANGES_REQUESTED** [review]
There should be a way to always align the checkmark on the first line of content, no matter the size of the checkmark or the text.

**Comment by alizedebray** [comment]
Waiting for the bullet list styles to be implemented with [#3437](https://github.com/swisspost/design-system/issues/3437)

**Pull request closed without merging**